src/sonardevices/Sonar.cpp

```diff
@@ -0,0 +1,100 @@
+
+#include <vector>
+#include <cstring>
+#include <memory.h>
+#include <sonar_driver/ezsocket/ezsocket.hxx>
+#include <sonar_driver/sonardevices/Sonar.h>
+#include <sonar_driver/sonardevices/OculusMessages.h>
+
+#include <iostream>
+
+using namespace EZSocket;
+
+Sonar::Sonar(){
+    fireMode = 1;
+    currRange = 20.0;
+    currGain = 1.0;
+    speedOfSound = 0.0;
+    salinity = 0.0;
+    temperature = 0.0;
+    pressure = 0.0;
+    gainAssistActive = false;
+    gamma = 0;
+    netSpeedLimit = 255;
+    pingRate = 0;
+    callbackThreadStarted = false;
+    callbackThreadActive = false;
+    state = SonarState::Ready;
+}
+
+Sonar::Sonar(std::shared_ptr<sensor_msgs::msg::Image> messagePointer) : Sonar(){
+    rosImageMessagePointer = messagePointer;
```

Comment:
@EulersBoiler The sonar has no business caring about ROS messages or ROS in general. This should be removed from the Sonar classes.